RoddyCore/src/de/dkfz/roddy/core/CohortDataRuntimeServiceExtension.groovy

```diff
@@ -114,6 +118,13 @@ class CohortDataRuntimeServiceExtension {
     List<DataSet> loadCohortDatasetsWithFilter(Analysis analysis, List<String> pidFilters) {
         List<DataSet> listOfDataSets = runtimeService.getListOfPossibleDataSets(analysis)
 
+        if (checkIfListOnlyContainsSCIdentifiers(pidFilters)) {
+            validateSuperCohortCValuesOrFail(analysis.configuration, pidFilters)
```

Comment:
No change necessary, but just a suggestion. I tend to call methods that validate something (without side-effects) `assertSomething()`. The assert would also express that the method will throw. The "fail" is less specific.